examples/gno.land/p/demo/accesscontrol/accesscontrol_test.gno

```diff
@@ -0,0 +1,228 @@
+	// Simulate the administrator as the current caller
+	admin := testutils.TestAddress("admin")
+	std.TestSetOrigCaller(admin)
+
+	// Create roles with the administrator
+	roles := &Roles{Admin: admin}
+
+	// Create a new role with a new administrator address
+	newRoleName := "newRole"
+	newRole := roles.CreateRole(newRoleName)
+
+	// Check that the new role has been created correctly
+	if newRole.Name != newRoleName {
+		t.Fatalf("expected new role name to be '%s', got '%s'", newRoleName, newRole.Name)
+	}
+	if newRole.AdminRole != admin {
+		t.Fatalf("expected new role admin role to be %s, got %s", admin.String(), newRole.AdminRole.String())
+	}
+
+	// Simulate newAdmin as the current caller
+	std.TestSetOrigCaller(admin)
+
+	// Explicitly add the role to the new administrator to check functionality
+	newRole.GrantRole(admin)
+
+	// Check if the new role has been added to the holder
+	if !newRole.HasRole(admin) {
+		t.Fatalf("expected new role to be added to the holder")
+	}
+}
+
+// TestCallerIsAdmin verifies the CallerIsAdmin method.
+func TestCallerIsAdmin(t *testing.T) {
+	adminRole := testutils.TestAddress("admin-address")
+	roleData := NewRole("ExampleRole", adminRole)
+	// Call CallerIsAdmin with admin caller
+	std.TestSetOrigCaller(adminRole)
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("expected no panic, got %v", r)
+		}
+	}()
+	roleData.AssertCallerIsAdmin()
+	// Call CallerIsAdmin with non-admin caller
+	user := testutils.TestAddress("user-address")
+	std.TestSetOrigCaller(user)
+	defer func() {
+		if r := recover(); r == nil {
+			t.Fatalf("expected panic, got nil")
+		}
+	}()
+	roleData.AssertCallerIsAdmin()
+}
+
+// Testing the RevokeRole Method for a Non-Admin
+func TestRevokeRoleNonAdmin(t *testing.T) {
+	admin := testutils.TestAddress("admin")
+	user1 := testutils.TestAddress("user1")
+	user2 := testutils.TestAddress("user2")
+
+	// Create role data with the administrator
+	roleData := NewRole("admin", admin)
+
+	// Grant role to user1
+	std.TestSetOrigCaller(admin)
+	roleData.GrantRole(user1)
+	if !roleData.HasRole(user1) {
+		t.Fatalf("expected user1 to have role")
+	}
+
+	// Simulate user2 as the current caller
+	std.TestSetOrigCaller(user2)
+
+	// Attempting to revoke user1's role as user2 (non-admin)
+	defer func() {
+		if r := recover(); r == nil {
+			t.Fatalf("expected panic when non-admin tries to revoke role")
+		}
+	}()
+	roleData.RevokeRole(user1)
+}
+
+// Testing the RenounceRole method with Invalid Confirmation
+func TestRenounceRole(t *testing.T) {
+	adminRole := testutils.TestAddress("admin-address")
+	roleData := NewRole("ExampleRole", adminRole)
+	account := testutils.TestAddress("user-address")
+	// Simulate the administrator as the current caller
+	std.TestSetOrigCaller(adminRole)
+	// Grant role to the account
+	roleData.GrantRole(account)
+	// Simulate the account as the current caller
+	std.TestSetOrigCaller(account)
+	// Renounce the role
+	err := roleData.RenounceRole(account)
```

Comment:
`uassert.NoError`

Reply:
I refactored all the tests
- https://github.com/gnolang/gno/pull/2307/commits/e973d18551cbfe3d2e85dc74174f38259e27548a